File: fixtures/scale/dart/large/lib/ciphers/myszkowski/myszkowski_cipher.dart
import '../../core/cipher.dart';
import '../columnar/columnar_cipher.dart';
import '../columnar/columnar_key.dart';
import 'myszkowski_key.dart';

/// Myszkowski transposition with a repeated-letter keyword. Equal letters
/// read left to right, realised here by delegating to a plain columnar pass
/// over the tie-broken column order.
class MyszkowskiCipher implements Cipher {
  final ColumnarCipher _delegate;

  MyszkowskiCipher(MyszkowskiKey key)
      : _delegate = ColumnarCipher(ColumnarKey(key.keyword));

  @override
  String name() => 'myszkowski';

  @override
  String encrypt(String plaintext) => _delegate.encrypt(plaintext);

  @override
  String decrypt(String ciphertext) => _delegate.decrypt(ciphertext);
}

File: fixtures/scale/scala/large/src/cryptobench/ciphers/myszkowski/MyszkowskiCipher.scala
package cryptobench.ciphers.myszkowski

import cryptobench.ciphers.columnar.ColumnarCipher
import cryptobench.ciphers.columnar.ColumnarKey
import cryptobench.core.Cipher

/**
 * Myszkowski transposition with a repeated-letter keyword. Equal letters read
 * left to right, realised here by delegating to a plain columnar pass over
 * the tie-broken column order.
 */
final class MyszkowskiCipher(key: MyszkowskiKey) extends Cipher {
  private val delegate = new ColumnarCipher(ColumnarKey(key.keyword))

  override def name: String = "myszkowski"

  override def encrypt(plaintext: String): String = delegate.encrypt(plaintext)

  override def decrypt(ciphertext: String): String = delegate.decrypt(ciphertext)
}
